src/network-area-diagram-viewer.ts

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { SVG } from '@svgdotjs/svg.js';
+import '@svgdotjs/svg.panzoom.js';
+
+type DIMENSIONS = { width: number; height: number; viewbox: VIEWBOX };
+type VIEWBOX = { x: number; y: number; width: number; height: number };
+
+export class NetworkAreaDiagramViewer {
+    container: HTMLElement;
+    svgContent: string;
+    width: number;
+    height: number;
+
+    constructor(
+        container: HTMLElement,
+        svgContent: string,
+        minWidth: number,
+        minHeight: number,
+        maxWidth: number,
+        maxHeight: number
+    ) {
+        this.container = container;
+        this.svgContent = svgContent;
+        this.init(minWidth, minHeight, maxWidth, maxHeight);
+    }
+
+    public setWidth(width: number): void {
+        this.width = width;
+    }
+
+    public setHeight(height: number): void {
+        this.height = height;
+    }
+
+    public setContainer(container: HTMLElement): void {
+        this.container = container;
+    }
+
+    public setSvgContent(svgContent: string): void {
+        this.svgContent = svgContent;
+    }
+
+    public getWidth(): number {
+        return this.width;
+    }
+
+    public getHeight(): number {
+        return this.height;
+    }
+
+    public getContainer(): HTMLElement {
+        return this.container;
+    }
+
+    public getSvgContent(): string {
+        return this.svgContent;
+    }
+
+    public init(
+        minWidth: number,
+        minHeight: number,
+        maxWidth: number,
+        maxHeight: number
+    ): void {
+        if (
+            this.container === null ||
```

Comment:
if (this.container && this.this.svgContent)

Reply:
done